frontend/src/components/@common/AnimationDiv/AnimationDiv.tsx

```diff
@@ -0,0 +1,120 @@
+import { useState, useLayoutEffect, HTMLAttributes, AnimationEventHandler } from 'react';
+import { css, styled } from 'styled-components';
+import * as animation from 'styles/animation';
+
+type DisplayState = 'hidden' | 'appear' | 'disappear' | 'show';
+type Animation = 'slide';
+type DirectionTo = 'up' | 'down' | 'left' | 'right';
+
+type Props = {
+  appearAnimation?: Animation;
+  disappearAnimation?: Animation;
+  animationDuration?: number;
+  directionTo?: DirectionTo;
+  isVisible: boolean;
+  children: React.ReactNode;
+  onAppear?: AnimationEventHandler<HTMLDivElement>;
+  onDisappear?: AnimationEventHandler<HTMLDivElement>;
+} & Omit<HTMLAttributes<HTMLDivElement>, 'onAnimationEnd'>;
+
+// hidden -> appear(애니메이션) -> show -> disappear(애니메이션) -> hidden 반복.
+const AnimationDiv = ({
+  appearAnimation = 'slide',
+  disappearAnimation = 'slide',
+  animationDuration = 0.5,
+  directionTo = 'up',
+  isVisible,
+  children,
+  onAppear,
+  onDisappear,
+  ...rest
+}: Props) => {
+  const [displayState, setDisplayState] = useState<DisplayState>('hidden');
+
+  // 바깥에서 받아온 isVisible 상태로 appear할 건지 disappear 할 건지 결정.
+  // isVisible을 true로 받으면 appear, display가 show인 상태에서 isVisible을 false로 받으면 disappear.
+  useLayoutEffect(
+    function checkDisplayState() {
+      const isConditionAppear = isVisible;
+      const isConditionDisappear = displayState === 'show' && !isVisible;
+
+      if (!isConditionAppear && !isConditionDisappear) return;
+
+      setDisplayState(isVisible ? 'appear' : 'disappear');
+    },
+    [isVisible],
+  );
```

Comment:
`function` 을 사용하신 이유가 궁금해용

Reply:
화살표 함수로 쓰면 다음과 같은데 하는 동작을 함수명으로 명시하는 목적이 컸는데 흠 이 부분은 그냥 일관성 있게 화살표 함수로 변경하였습니다.
```suggestion
  useLayoutEffect(() => {
      const isConditionAppear = isVisible;
      const isConditionDisappear = displayState === 'show' && !isVisible;

      if (!isConditionAppear && !isConditionDisappear) return;

      setDisplayState(isVisible ? 'appear' : 'disappear');
    },
    [isVisible],
  );
```